src/main/java/software/amazon/awssdk/crt/utils/StringUtils.java

```diff
@@ -23,4 +23,58 @@ public static String join(CharSequence delimiter, Iterable<? extends CharSequenc
         }
         return sb.toString();
     }
+
+    /**
+     * Encode a byte[] array into a Base64 String in pure Java
+     */
+    private final static char[] BASE64_VALID_CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/".toCharArray();
+    public static String simpleBase64Encode(byte[] data) {
```

Comment:
‼️why can't we just use Java's built-in [Base64.Encoder](https://docs.oracle.com/javase/8/docs/api/java/util/Base64.Encoder.html)?

Reply:
See: https://github.com/awslabs/aws-crt-java/pull/552#discussion_r1023361516

Summary: We cannot use `Base64.Encoder` because we currently have the minimum Android API level set to 24, which does not support all of Java 8 (it is really Java 7 with some additions). API Level 26+ supports Java 8 and `Base64`, but that would be dropping roughly 5% of Android users based on 2022 statistics.

There are other built-in solutions outside of `Base64`, but:
* `javax.xml.bind.DatatypeConverter` works for Base64 support for API level 24, but was dropped in Java version 11 onward, where it was dropped completely in favor of `Base64`.
* `android.util.Base64` works for Android all the way back to API level 8, but is not included in standard Java. I could also not find a way to say "If in Android, use android.utils.Base64, otherwise use Java 8's Base64".
* Appache Commons Codec has a Base64 implementation, but you have to pull in their entire conversion library. Seems heavy for just Base64 encoding for just the Java keystore. Also unsure on license for library.

So making a Base64 encoder in pure Java without using any modules fixes this issue, allowing Android with API level 24 to work and all the versions of Java we support. It is not ideal, ideally we'd just drop Android API Level 24 for Android and then use `Base64`, but dropping roughly 5% of the Android devices just to support the Java keystore seems like a bad tradeoff.

Reply:
```
/*
 * Base 64 encodes the contents of to_encode and stores the result in output.
 */
AWS_COMMON_API
int aws_base64_encode(const struct aws_byte_cursor *AWS_RESTRICT to_encode, struct aws_byte_buf *AWS_RESTRICT output);
```
[https://github.com/awslabs/aws-c-common/blob/03d1b0e83094d1168b70b007f998b6a52ae507cf/include/aws/common/encoding.h#L69